sympy/geometry/parabola.py

```diff
@@ -47,32 +49,77 @@ class Parabola(GeometrySet):
     ValueError
         When `focus` is not a two dimensional point.
         When `focus` is a point of directrix.
+        When more than one variable has name `x` or `y` in `eq` or `eq` is not the equation of a vertical or horizontal parabola.
     NotImplementedError
         When `directrix` is neither horizontal nor vertical.
 
     Examples
     ========
 
-    >>> from sympy import Parabola, Point, Line
-    >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7,8)))
+    >>> from sympy import Parabola, Point, Line, pi
+    >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
     >>> p1.focus
     Point2D(0, 0)
     >>> p1.directrix
     Line2D(Point2D(5, 8), Point2D(7, 8))
 
-    """
+    You can find the intersection of a Parabola with an axis of interest:
 
-    def __new__(cls, focus=None, directrix=None, **kwargs):
+    >>> from sympy.abc import x, y
+    >>> y_axis = Line(x)
+    >>> x_axis = Line(y)
+    >>> p1.intersection(y_axis)
+    [Point2D(0, 4)]
+    >>> p1.intersection(x_axis)
+    [Point2D(-8, 0), Point2D(8, 0)]
 
-        if focus:
-            focus = Point(focus, dim=2)
-        else:
-            focus = Point(0, 0)
+    The Parabola may be described by passing coefficients
+    from the quadratic expression:
+
+    >>> Parabola(abc=(1, 2, 3))  # x**2 + 2*x + 3
+    Parabola(Point2D(-1, 9/4), Line2D(Point2D(0, 7/4), Point2D(1, 7/4)))
 
-        directrix = Line(directrix)
+    If the coefficients were for a horizontally-oriented Parabola, then
+    a clockwise rotation can be used:
 
-        if directrix.contains(focus):
-            raise ValueError('The focus must not be a point of directrix')
+    >>> p2 = _.rotate(-pi/2)  # th = -pi/2
+    Parabola(Point2D(9/4, -1), Line2D(Point2D(7/4, 0), Point2D(7/4, 1)))
+
+    Every parabola is a translated, scaled (and potentially rotated)
+    version of the basic ``y = x^2`` parabola:
+
+    >>> p2.equation()
+    x - y**2 - 2*y - 3
+    >>> v = p2.vertex
+    >>> a = 4*p2.p_parameter
+    >>> th = -pi/2
+    >>> p2.translate(*-v).rotate(-th).scale(y=1/a).equation()
+    -x**2 + y
+    >>> p2.is_similar(Parabola())
+    True
+
+    """
+
+    def __new__(cls, focus=None, directrix=None, abc=None, **kwargs):
+
+        if directrix is not None:
+            if abc is not None:
+                raise ValueError('enter focus & directrix or abc')
+            focus = Point(focus or (0, 0), dim=2)
+            directrix = Line(directrix)
+            if directrix.contains(focus):
+                raise ValueError('The focus must not be a point of directrix')
+        else:
+            if focus is not None:
+                raise ValueError('enter focus & directrix or abc')
```

Comment:
I think that the specific reason `a signature is made more complex` is that revealed here.

For example, you are branching first with `directrix` and then 
in the nested if statement, you are writing the code that assumes something eliminate the logically negative cases (`directix is None`)

So it is literally the case
```
if focus is not None and directrix is not None and abc is None:
    ...
if focus is None and directrix is None and abc is not None:
    ...
```

Of course, it may not be that much difficult to understand this at the point now,
but later modifying/understanding the code should be equivalent to solving some satisfiability problem about the logics here, and that's one thing I'd like to comment about here.

Reply:
That's a good point and I wonder if you raised that some years ago. Every decision in a routine adds to the complexity both of testing and "digesting". That's good enough reason to go for a dedicated `abc` method.